tests/cross-compile/dxc-error.hlsl.expected

```diff
@@ -0,0 +1,8 @@
+result code = -1
+standard error = {
+dxc: slang:9:2: error: use of undeclared identifier 'gOutputBuffer'
```

Comment:
We should probably fix up the path that we pass to dxc, to at least match what we pass to fxc for the pass-through case.

In the non-pass-through case we might want to have the nominal file name be `slang-generated` or similar. (And bonus points would be if when dumping of intermediate results is on we emit the path of the intermediate file we dumped...)

Reply:
Added support to output 'slang-generated' or the source filename/s depending on if it's pass through.